### PR TITLE
managed: machine upgrade guide

### PR DESCRIPTION
**Pull request by bobheadxi**
Creates a guide, partially leveraging the existing tooling, for upgrading the machine for a managed instance.

Closes https://github.com/sourcegraph/customer/issues/481 , follow-up to https://github.com/sourcegraph/deploy-sourcegraph-managed/pull/96#issuecomment-915613916 , required to resolve https://github.com/sourcegraph/customer/issues/478

The first commit isn't worth reviewing (just misc. readability and structure changes) - the bulk of the changes is in https://github.com/sourcegraph/about/commit/882cd70baa7b59b6fae426d1c132a52e3b219f43

### CODE REVIEW

File: handbook/engineering/distribution/managed/upgrade_process.md [hunk 1]
```diff
@@ -245,48 +258,190 @@ During this time users will see 500 errors:
...
+
+Set up configuration for the new machine:
+
+```sh
+cp -R $OLD_DEPLOYMENT/ $NEW_DEPLOYMENT/
+git add $NEW_DEPLOYMENT/ && git commit -m "$CUSTOMER: set up $NEW_DEPLOYMENT configuration"
+```
+
+Initialize the new production deployment (`NEW_DEPLOYMENT`) using the snapshot created in the previous step.
+**⚠️ When the tool prompts you to apply, exit the command (DO NOT APPLY THE CHANGES). ⚠️**
```
Comment (jhchabran) [inline handbook/engineering/distribution/managed/upgrade_process.md:372]: I'm wondering, would it be possible to fix that with a flag, like making https://github.com/sourcegraph/deploy-sourcegraph-managed/blob/main/util/init-deployment.ts#L30 conditional to that flag? That'd make this step less scary 😅 !

**Review by jhchabran — COMMENTED** [review]
All good documentation wise for me 👍  I am not approving here because I don't have enough XP on the infra part to review the commands thoroughly. 

Added a minor comment with a potential low hanging fruit.

**Review by davejrt — APPROVED** [review]
LGTM

File: handbook/engineering/distribution/managed/upgrade_process.md (hunk 1 above)
Comment (bobheadxi) [reply handbook/engineering/distribution/managed/upgrade_process.md:372]: I think the terraform command stopping to ask you for a `yes` confirmation is pretty close to the effectiveness of such a flag. This is the only place it should happen, and machine upgrades happen rarely (and presumably by someone who's done a normal upgrade before so is familiar), so I think i'd rather avoid potentially missing the apply in the normal upgrade process - WDYT?

**Review by daxmc99 — APPROVED** [review]
LGTM!

**Pull request merged**